run_testlist.py

```diff
@@ -17,266 +17,424 @@
+                  # Number of servers, number of clients, timeout in minutes
+                  'test_plan': [
+                      (4, 1, 15),
+                      (4, 1, 15),
+                      (8, 1, 15),
+                      (16, 1, 15),
+                      (32, 1, 15),
+                      (64, 1, 15),
+                      (128, 1, 15),
+                      (256, 1, 15),
+                      (512, 1, 15),
+                      (1024, 1, 15)
+                  ],
+                  'ppc': 32,
+                  'env_vars': {
+                      'pool_size': '85G'
+                  },
+                  'enabled': False
+                  }
+                 ]
+
+
+class TestList(object):
+    def __init__(self, test_group, testlist, env, script='run_sbatch.sh'):
+        self._test_group = test_group
+        self._testlist = testlist
+        self._env = env
+        self._teardown_offset = 10
+        dst_dir = os.getenv('DST_DIR')
+        self._script = os.path.join(dst_dir, script)
+
+    def _expand_default_env_vars(self, env, test):
+        env['TEST_GROUP'] = self._test_group
         env['TESTCASE'] = test['testcase']
-        for i in range(len(test['nServer'])):
-            srv = test['nServer'][i]
-            cli = test['nClient'][i]
-            nodes = srv + cli + 1
-            cores = nodes * test['ppc']
-            if nodes <= 512:
-                env['PARTITION'] = 'normal'
-            else:
-                env['PARTITION'] = 'large'
-
-            env['DAOS_SERVERS'] = str(srv)
-            env['DAOS_CLIENTS'] = str(cli)
-            env['NNODE'] = str(nodes)
-            env['NCORE'] = str(cores)
-            env['PPC'] = str(test['ppc'])
-            env['POOL_SIZE'] = test['pool_sz']
-            env['N_FILE'] = test['nFile']
-            env['BYTES_READ'] = test['bytes_read']
-            env['BYTES_WRITE'] = test['bytes_write']
-            env['TREE_DEPTH'] = test['tree_depth']
-
-            t = test['timeout'][i] + 10
-            h = int(t / 60)
-            m = t % 60
-            s = 0
-            env['TIMEOUT'] = str(h) + ":" + str(m) + ":" + str(s)
-            env['OMPI_TIMEOUT'] = str(test['timeout'][i] * 60)
-
-            subprocess.Popen(script, env=env)
+        env['PPC'] = str(test['ppc'])
+
+    def _expand_extra_end_vars(self, env, test):
+        env_vars = test.get('env_vars', {})
+        for name, value in env_vars.items():
+            env[name.upper()] = str(value)
+
+    def _add_partition(self, env, nodes):
+        if nodes <= 512:
+            env['PARTITION'] = 'normal'
+        else:
+            env['PARTITION'] = 'large'
+
+    def _add_timeout(self, env, timeout):
+        h = (timeout + self._teardown_offset) // 60
+        m = (timeout + self._teardown_offset) % 60
+        s = 0
+        env['TIMEOUT'] = str(h) + ":" + str(m) + ":" + str(s)
+        env['OMPI_TIMEOUT'] = str(timeout * 60)
+
+    def _expand_scenario(self, env, ppc, scenario):
```

Comment:
let's rename 'scenario' here to testcase
def _expand_testcase(self, env, ppc, testcase):